File: initial-code/Client-Java/src/main/java/pt/ulisboa/tecnico/tuplespaces/client/ClientMain.java
package pt.ulisboa.tecnico.tuplespaces.client;

import pt.ulisboa.tecnico.tuplespaces.client.grpc.ClientService;

public class ClientMain {

    public static void main(String[] args) {

        System.out.println(ClientMain.class.getSimpleName());

        // receive and print arguments
        System.out.printf("Received %d arguments%n", args.length);
        for (int i = 0; i < args.length; i++) {
            System.out.printf("arg[%d] = %s%n", i, args[i]);
        }

        // check arguments
        if (args.length != 2) {
            System.err.println("Argument(s) missing!");
            System.err.println("Usage: mvn exec:java -Dexec.args=<host:port> <client_id>");
            return;
        }

        // get the host and the port of the server or front-end
        final String host_port = args[0];
        final int client_id = Integer.parseInt(args[1]);

        CommandProcessor parser = new CommandProcessor(new ClientService(host_port, client_id));
        parser.parseInput();

    }
}

File: initial-code/Client-Python/client_main.py
import sys
from typing import List
from client_service import ClientService
from command_processor import CommandProcessor

class ClientMain:
    @staticmethod
    def main(args: List[str]):
        print("Python Client")

        # receive and print arguments
        print(f"Received {len(args)} arguments")
        for i, arg in enumerate(args):
            print(f"arg[{i}] = {arg}")

        # check arguments
        if len(args) < 2:
            print("Argument(s) missing!", file=sys.stderr)
            print("Usage: python3 client_main.py <host:port> <client_id>", file=sys.stderr)
            return

        # Get the host and port of the server or front-end
        host_port = args[0]
        client_id = args[1]

        parser = CommandProcessor(ClientService(host_port, client_id))
        parser.parse_input()


if __name__ == "__main__":
    ClientMain.main(sys.argv[1:])